src/Button/Button.component.tsx

```diff
@@ -370,15 +375,6 @@ const StyledButton = styled('button')<StyledButtonProps>`
     align-items: center;
     outline: none !important;
 
-    margin: ${({ margin }: StyledButtonProps) => margin || `0`};
-    ${({ block, margin }: StyledButtonProps) =>
-        !margin &&
-        css`
-            & + .anchor-button {
```

Comment:
This was designed to automatically space buttons that are side by side. In practice I think it caused more surprise/frustration than it solved, and with space props added we can handle these cases easily/explicitly.